streets_reader/src/lib.rs

```diff
@@ -84,6 +86,14 @@ fn extract_osm(
         timer.next();
         out.handle_way(*id, way, &streets.config);
     }
+    timer.start_iter(
+        "processing OSM ways split into pieces",
+        doc.clipped_copied_ways.len(),
+    );
+    for (id, way) in &doc.clipped_copied_ways {
```

Comment:
What's up with these? The problem is this:
![Screenshot from 2022-12-15 16-10-56](https://user-images.githubusercontent.com/1664407/207915156-654e375b-e2bb-434c-8415-d369d1aed61b.png)
We've got a way that crosses the boundary twice. In an earlier version of this PR, we got this incorrect result. Now we split this one way into 3 pieces, discard the middle one (out of bounds). And thankfully multiple ways per OSM ID don't cause any headache elsewhere, since we're already using things like opaque IDs.